checks.d/mysql.py

```diff
@@ -32,78 +33,341 @@
+    'Innodb_ibuf_merged_inserts': ('mysql.innodb.ibuf_merged_inserts', RATE),
+    'Innodb_ibuf_merges': ('mysql.innodb.ibuf_merges', RATE),
+    'Innodb_ibuf_segment_size': ('mysql.innodb.ibuf_segment_size', GAUGE),
+    'Innodb_ibuf_size': ('mysql.innodb.ibuf_size', GAUGE),
+    'Innodb_lock_structs': ('mysql.innodb.lock_structs', RATE),
+    'Innodb_locked_tables': ('mysql.innodb.locked_tables', GAUGE),
+    'Innodb_locked_transactions': ('mysql.innodb.locked_transactions', GAUGE),
+    'Innodb_log_waits': ('mysql.innodb.log_waits', RATE),
+    'Innodb_log_write_requests': ('mysql.innodb.log_write_requests', RATE),
+    'Innodb_log_writes': ('mysql.innodb.log_writes', RATE),
+    'Innodb_lsn_current': ('mysql.innodb.lsn_current', RATE),
+    'Innodb_lsn_flushed': ('mysql.innodb.lsn_flushed', RATE),
+    'Innodb_lsn_last_checkpoint': ('mysql.innodb.lsn_last_checkpoint', RATE),
+    'Innodb_mem_adaptive_hash': ('mysql.innodb.mem_adaptive_hash', GAUGE),
+    'Innodb_mem_additional_pool': ('mysql.innodb.mem_additional_pool', GAUGE),
+    'Innodb_mem_dictionary': ('mysql.innodb.mem_dictionary', GAUGE),
+    'Innodb_mem_file_system': ('mysql.innodb.mem_file_system', GAUGE),
+    'Innodb_mem_lock_system': ('mysql.innodb.mem_lock_system', GAUGE),
+    'Innodb_mem_page_hash': ('mysql.innodb.mem_page_hash', GAUGE),
+    'Innodb_mem_recovery_system': ('mysql.innodb.mem_recovery_system', GAUGE),
+    'Innodb_mem_thread_hash': ('mysql.innodb.mem_thread_hash', GAUGE),
+    'Innodb_mem_total': ('mysql.innodb.mem_total', GAUGE),
+    'Innodb_os_file_fsyncs': ('mysql.innodb.os_file_fsyncs', RATE),
+    'Innodb_os_file_reads': ('mysql.innodb.os_file_reads', RATE),
+    'Innodb_os_file_writes': ('mysql.innodb.os_file_writes', RATE),
+    'Innodb_os_log_pending_fsyncs': ('mysql.innodb.os_log_pending_fsyncs', RATE),
+    'Innodb_os_log_pending_writes': ('mysql.innodb.os_log_pending_writes', RATE),
+    'Innodb_os_log_written': ('mysql.innodb.os_log_written', RATE),
+    'Innodb_pages_created': ('mysql.innodb.pages_created', RATE),
+    'Innodb_pages_read': ('mysql.innodb.pages_read', RATE),
+    'Innodb_pages_written': ('mysql.innodb.pages_written', RATE),
+    'Innodb_pending_aio_log_ios': ('mysql.innodb.pending_aio_log_ios', RATE),
+    'Innodb_pending_aio_sync_ios': ('mysql.innodb.pending_aio_sync_ios', RATE),
+    'Innodb_pending_buffer_pool_flushes': ('mysql.innodb.pending_buffer_pool_flushes', RATE),
+    'Innodb_pending_checkpoint_writes': ('mysql.innodb.pending_checkpoint_writes', RATE),
+    'Innodb_pending_ibuf_aio_reads': ('mysql.innodb.pending_ibuf_aio_reads', RATE),
+    'Innodb_pending_log_flushes': ('mysql.innodb.pending_log_flushes', RATE),
+    'Innodb_pending_log_writes': ('mysql.innodb.pending_log_writes', RATE),
+    'Innodb_pending_normal_aio_reads': ('mysql.innodb.pending_normal_aio_reads', RATE),
+    'Innodb_pending_normal_aio_writes': ('mysql.innodb.pending_normal_aio_writes', RATE),
```

Comment:
All these `pending` metrics should probably be sent as gauges (if they're actually giving the the number of pending operations at a given moment).

https://www.percona.com/blog/2006/07/17/show-innodb-status-walk-through/ describes these metrics, according to it they should indeed be sent as gauges.